src/CFamily.UnitTests/CMake/ActiveConfigProviderTests.cs

```diff
@@ -0,0 +1,120 @@
+        }
+
+        [TestMethod]
+        public void Get_NullRootDirectory_Throws()
+        {
+            var testSubject = new ActiveConfigProvider(Mock.Of<ILogger>());
+
+            Action act = () => testSubject.GetActiveConfig(null);
+            act.Should().ThrowExactly<ArgumentNullException>().And.ParamName.Should().Be("rootDirectory");
+        }
+
+        [TestMethod]
+        public void Get_MissingSettingsFile_ReturnsDefaultConfig()
+        {
+            const string rootDir = "c:\\aaa";
+            var fullSettingsPath = CalcFullSettingsPath(rootDir);
+
+            var fileSystem = CreateFileSystemWithNoFiles();
+            var logger = new TestLogger(logToConsole: true);
+            var testSubject = new ActiveConfigProvider(logger, fileSystem.Object);
+
+            testSubject.GetActiveConfig(rootDir).Should().Be(ExpectedDefaultConfig);
+
+            AssertSettingsFileExistenceChecked(fileSystem, fullSettingsPath);
+        }
+
+        [TestMethod]
+        [DataRow("not valid json", ExpectedDefaultConfig)]
+        [DataRow("{ 'SettingIsMissing' : 'aaa'}", ExpectedDefaultConfig)]
+        [DataRow("{ 'CurrentProjectSetting': 'my config' }", "my config")]
+        public void Get_FileExists_ReturnsCorrectConfig(string settingsJson, string expectedConfig)
+        {
+            const string rootDir = "c:\\xxx";
+            var fullSettingsPath = CalcFullSettingsPath(rootDir);
+
+            var fileSystem = CreateFileSystemWithFile(fullSettingsPath, settingsJson);
+            var logger = new TestLogger(logToConsole: true);
+            var testSubject = new ActiveConfigProvider(logger, fileSystem.Object);
+
+            testSubject.GetActiveConfig(rootDir).Should().Be(expectedConfig);
+
+            AssertSettingsFileExistenceChecked(fileSystem, fullSettingsPath);
+            AssertSettingsFileRead(fileSystem, fullSettingsPath);
+        }
+
+        private static Mock<IFileSystem> CreateFileSystemWithNoFiles()
+        {
+            var fileSystem = new Mock<IFileSystem>();
+            fileSystem.Setup(x => x.File.Exists(It.IsAny<string>())).Returns(false);
+            return fileSystem;
+        }
+
+        private static Mock<IFileSystem> CreateFileSystemWithFile(string fullPath, string content)
+        {
+            var fileSystem = new Mock<IFileSystem>();
+            fileSystem.Setup(x => x.File.Exists(fullPath)).Returns(true);
+            fileSystem.Setup(x => x.File.ReadAllText(fullPath)).Returns(content);
+            return fileSystem;
+        }
+
+        private static void AssertSettingsFileExistenceChecked(Mock<IFileSystem> fileSystem, string fullSettingsPath)
+        {
+            fileSystem.Verify(x => x.File.Exists(fullSettingsPath), Times.Once);
+        }
+
+        private static void AssertSettingsFileRead(Mock<IFileSystem> fileSystem, string rootDirectory)
+        {
+            var fullPath = CalcFullSettingsPath(rootDirectory);
```

Comment:
`fullPath` is not used and `rootDirectory` is in fact `fullPath` (passed by the caller), I'm guessing it's leftovers from refactor.